### PR TITLE
4273: Added a new create method to pass offset and length of sub array.

### LINKED ISSUE #4273: In 'google-cloud-storage', Allow passing data size to Storage.create
Thanks for stopping by to let us know something could be better!

Please include as much information as possible:

#### Environment details

- OS:
- Java version:
- google-cloud-java version(s):

#### Steps to reproduce

1. ?
2. ?

#### Stacktrace

```
Any relevant stacktrace here.
```

#### Code snippet

```java
Any relevant code snippet to help reproduce the issue.
```

#### External references such as API reference guides used

- ?

#### Any additional information below

Following these steps will guarantee the quickest resolution possible.

Thanks!

### PR DESCRIPTION
**Pull request by andrey-qlogic**
Fixes #4273

### CODE REVIEW

**Comment by andrey-qlogic** [comment]
@JesseLovelace, please take a look at this PR

**Comment by sduskis** [comment]
@frankyn, PTAL

File: google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java [hunk 1]
```diff
@@ -132,6 +132,19 @@ public Blob createBlobFromByteArray(String bucketName, String blobName) {
     return blob;
   }
 
+  /** Example of creating a blob with sub array from a byte array. */
+  // [TARGET create(BlobInfo, byte[], offset, length, BlobTargetOption...)]
+  // [VARIABLE "my_unique_bucket"]
+  // [VARIABLE "my_blob_name"]
+  public Blob createBlobWithSubArrayFromByteArray(String bucketName, String blobName) {
```
Comment (frankyn) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java]: Could you add an integration test for this snippet?

**Comment by sduskis** [comment]
@frankyn, can this PR be merged? (i.e. can your question be resolved offline?)

**Pull request merged**